### PR TITLE
Add the ability to provide a view spec to Helix Branches SCM

### PR DESCRIPTION
**Pull request by fbyrne**
Fixed pull request with unchanged files reverted.
Fixed default filter and mappings

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/p4/scm/BranchesScmSource.java [hunk 1]
```diff
@@ -97,10 +113,47 @@ public P4Browser getBrowser() {
...
+		StringBuffer workspaceView = new StringBuffer(1024);
+		workspaceView.append(String.format(mappingFormat, getScriptPathOrDefault("Jenkinsfile")));
+		for(String mapping : getViewMappings()){
+			workspaceView.append("\n").append(String.format(mappingFormat, mapping));
+		}
+		WorkspaceSpec spec = new WorkspaceSpec(workspaceView.toString(), null);
+		return new ManualWorkspaceImpl(getCharset(), false, client, spec);
+	}
+
+	private String getScriptPathOrDefault(String defaultScriptPath) {
```
Comment (p4paul) [inline src/main/java/org/jenkinsci/plugins/p4/scm/BranchesScmSource.java]: I moved getScriptPathOrDefault to the Abstract class AbstractP4ScmSource; to save duplication could you change it to protected in AbstractP4ScmSource and remove it from here.

**Comment by p4paul** [comment]
Sorry for the delay, I have been traveling and now at QCon.  Will try and get it in Main for you after the minor change.

**Comment by fbyrne** [comment]
Thanks Paul, Just wondering when the next release will be cut so we can start deploying to our ci's?

**Comment by p4paul** [comment]
I can release tomorrow if no bugs show up.  You can get the latest HPI build here:
https://ci.jenkins.io/job/Plugins/job/p4-plugin/job/master/129

**Pull request merged**